http/utils/PollRequestManager.js

```diff
@@ -9,9 +9,12 @@ function PollRequestManager(fetchFunction, pollingTimeout = 1000){
 		this.execute = function(){
 			//if there is a delayedStart and it's the first time when the request is executed
 			if(delayedStart && typeof currentState === "undefined"){
-				setTimeout(function(){
-					currentState = fetchFunction(url, options);
-				}, delayedStart);
+				return new Promise((resolve, reject) => {
```

Comment:
A promise needs to be returned in order to prevent "calling `then()` on undefined" down below